### PR TITLE
Fix #702: Add Profile Update Layout

### LINKED ISSUE #702: Lowfi - Add profile updated
Mock Link: https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/77b5658b-cd35-4eda-b971-f1f4fd0181de/PC-MP-Add-Profile-Empty-7

The current implementation in develop branch for `Add Profile` is different as compared to the new mocks.

Please update this. Also, write test cases for the same.

### PR DESCRIPTION
**Pull request by aggarwalpulkit596**
## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by aggarwalpulkit596** [comment]
@rt4914 Added test, squashed commits and renamed branch couldn't do on the previous PR

**Comment by aggarwalpulkit596** [comment]
ezgif com-video-to-gif

**Comment by rt4914** [comment]
> @rt4914 Added test, squashed commits and renamed branch couldn't do on the previous PR

Sounds great. Just one more thing, can you fix the background color issue at the bottom of the screen.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

File: app/src/main/res/layout/add_profile_activity.xml [hunk 1]
```diff
@@ -26,35 +27,66 @@
...
         android:layout_height="wrap_content"
-        app:layout_constraintTop_toBottomOf="@+id/upload_image_button"
+        app:layout_constraintTop_toBottomOf="@+id/required_heading"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintEnd_toEndOf="parent"
         app:label="@string/add_profile_input_name"
         profile:error="@{viewModel.nameErrorMsg}"/>
+
+
+
```
Comment (rt4914) [inline app/src/main/res/layout/add_profile_activity.xml]: Just one small change, the spacing and indentation in this file is incorrect.

`Control + Alt + L`

**Review by rt4914 — CHANGES_REQUESTED** [review]
One change regarding indentation.

**Comment by aggarwalpulkit596** [comment]
> > @rt4914 Added test, squashed commits and renamed branch couldn't do on the previous PR
> 
> Sounds great. Just one more thing, can you fix the background color issue at the bottom of the screen.

should I make it completely white/yellow ?

File: app/src/main/res/layout/add_profile_activity.xml (hunk 1 above)
Comment (aggarwalpulkit596) [reply app/src/main/res/layout/add_profile_activity.xml]: > Just one small change, the spacing and indentation in this file is incorrect.
> 
> `Control + Alt + L`

doing this caused a lot of changes

**Comment by rt4914** [comment]
> > > @rt4914 Added test, squashed commits and renamed branch couldn't do on the previous PR
> > 
> > 
> > Sounds great. Just one more thing, can you fix the background color issue at the bottom of the screen.
> 
> should I make it completely white/yellow ?

Make it same as mocks, which is yellow.

File: app/src/main/res/layout/add_profile_activity.xml (hunk 1 above)
Comment (rt4914) [reply app/src/main/res/layout/add_profile_activity.xml]: It should look something like this at the end, which is consistent with all other xml codes.
```
<?xml version="1.0" encoding="utf-8"?>
<layout xmlns:android="http://schemas.android.com/apk/res/android"
  xmlns:app="http://schemas.android.com/apk/res-auto"
  xmlns:profile="http://schemas.android.com/tools">

  <data>

    <import type="android.view.View" />

    <variable
      name="viewModel"
      type="org.oppia.app.profile.AddProfileViewModel" />
  </data>

  <ScrollView
    android:id="@+id/scroll"
    android:layout_width="match_parent"
    android:layout_height="match_parent">

    <androidx.constraintlayout.widget.ConstraintLayout
      android:layout_width="match_parent"
      android:layout_height="wrap_content"
      android:background="@color/addProfileBackground"
      android:paddingTop="40dp"
      android:paddingBottom="40dp">

      <de.hdodenhof.circleimageview.CircleImageView
        android:id="@+id/upload_image_button"
        android:layout_width="136dp"
        android:layout_height="136dp"
        android:paddingBottom="8dp"
        android:src="@drawable/ic_upload_photo"
        app:layout_constraintEnd_toEndOf="parent"
        app:layout_constraintStart_toStartOf="parent"
        app:layout_constraintTop_toTopOf="parent" />

      <TextView
        android:id="@+id/required_heading"
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        android:layout_marginStart="32dp"
        android:layout_marginEnd="32dp"
        android:text="*Required"
        android:textSize="12sp"
        app:layout_constraintTop_toBottomOf="@+id/upload_image_button" />

      <org.oppia.app.profile.ProfileInputView
        android:id="@+id/input_name"
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        app:label="@string/add_profile_input_name"
        app:layout_constraintEnd_toEndOf="parent"
        app:layout_constraintStart_toStartOf="parent"
        app:layout_constraintTop_toBottomOf="@+id/required_heading"
        profile:error="@{viewModel.nameErrorMsg}" />

      <CheckBox
        android:id="@+id/checkbox_pin"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_marginStart="24dp"
        android:text="@string/add_profile_create_a_3_digit_pin"
        app:layout_constraintStart_toStartOf="parent"
        app:layout_constraintTop_toBottomOf="@+id/input_name" />

      <ImageView
        android:id="@+id/info_icon"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_marginTop="4dp"
        android:layout_marginEnd="40dp"
        android:src="@drawable/ic_info_icon"
        app:layout_constraintEnd_toEndOf="parent"
        app:layout_constraintTop_toTopOf="@+id/checkbox_pin" />

      <androidx.constraintlayout.widget.ConstraintLayout
        android:id="@+id/pin_container"
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        android:visibility="@{viewModel.createPin ? View.VISIBLE : View.GONE}"
        app:layout_constraintTop_toBottomOf="@+id/checkbox_pin">

        <org.oppia.app.profile.ProfileInputView
          android:id="@+id/input_pin"
          android:layout_width="match_parent"
          android:layout_height="wrap_content"
          android:layout_marginStart="@dimen/margin_32"
          android:visibility="visible"
          app:inputLength="3"
          app:isPasswordInput="true"
          app:label="@string/add_profile_input_pin"
          app:layout_constraintEnd_toEndOf="parent"
          app:layout_constraintStart_toStartOf="parent"
          app:layout_constraintTop_toTopOf="parent"
          profile:error="@{viewModel.pinErrorMsg}" />

        <org.oppia.app.profile.ProfileInputView
          android:id="@+id/input_confirm_pin"
          android:layout_width="match_parent"
          android:layout_height="wrap_content"
          android:layout_marginStart="@dimen/margin_32"
          app:inputLength="3"
          app:isPasswordInput="true"
          app:label="@string/add_profile_input_confirm_pin"
          app:layout_constraintEnd_toEndOf="parent"
          app:layout_constraintStart_toStartOf="parent"
          app:layout_constraintTop_toBottomOf="@+id/input_pin"
          profile:error="@{viewModel.confirmPinErrorMsg}" />

        <androidx.constraintlayout.widget.ConstraintLayout
          android:id="@+id/allow_download_container"
          android:layout_width="match_parent"
          android:layout_height="wrap_content"
          android:layout_marginStart="64dp"
          android:layout_marginTop="32dp"
          android:layout_marginEnd="32dp"
          app:layout_constraintTop_toBottomOf="@+id/input_confirm_pin">

          <LinearLayout
            android:layout_width="0dp"
            android:layout_height="wrap_content"
            android:layout_marginEnd="32dp"
            android:orientation="vertical"
            app:layout_constraintEnd_toStartOf="@+id/allow_download_switch"
            app:layout_constraintStart_toStartOf="parent"
            app:layout_constraintTop_toTopOf="parent">

            <TextView
              android:id="@+id/allow_download_heading"
              android:layout_width="wrap_content"
              android:layout_height="wrap_content"
              android:text="@string/add_profile_allow_download_heading"
              android:textColor="@{viewModel.validPin ? @color/black : @color/grey}"
              android:textStyle="bold" />

            <TextView
              android:id="@+id/allow_download_sub"
              android:layout_width="match_parent"
              android:layout_height="wrap_content"
              android:text="@string/add_profile_allow_download_sub" />
          </LinearLayout>

          <Switch
            android:id="@+id/allow_download_switch"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:clickable="@{viewModel.validPin ? true : false}"
            app:layout_constraintEnd_toEndOf="parent"
            app:layout_constraintTop_toTopOf="parent" />
        </androidx.constraintlayout.widget.ConstraintLayout>
      </androidx.constraintlayout.widget.ConstraintLayout>

      <Button
        android:id="@+id/create_button"
        style="@style/StateButtonActive"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_marginTop="60dp"
        android:layout_marginEnd="32dp"
        android:text="@string/add_profile_create_btn"
        android:textAllCaps="true"
        app:layout_constraintEnd_toEndOf="parent"
        app:layout_constraintTop_toBottomOf="@+id/pin_container" />
    </androidx.constraintlayout.widget.ConstraintLayout>
  </ScrollView>
</layout>

```

**Review by rt4914 — CHANGES_REQUESTED** [review]
PTAL

**Comment by aggarwalpulkit596** [comment]
@rt4914 updated 👍

**Comment by rt4914** [comment]
> @rt4914 updated 👍

Screenshot 2020-02-25 at 10 53 41 AM

@aggarwalpulkit596 We do not accept any force push commits. Always create a separate commit instead of force push on previous commit. This is done because force push changes the entire commit history which should not be done. I believe you will need to recreate this entire PR on a new branch and close this one.

**Comment by aggarwalpulkit596** [comment]
Okay will keep that in mind for future PR's other than this all good or do i need make some more changes

**Comment by rt4914** [comment]
> Okay will keep that in mind for future PR's other than this all good or do i need make some more changes

Other than that it looks good. Also, join: https://gitter.im/oppia/oppia-android

**Pull request closed without merging**